posthog/async_migrations/migrations/0008_cohort_actors.py

```diff
@@ -0,0 +1,89 @@
+from functools import cached_property
+
+from posthog.async_migrations.definition import AsyncMigrationDefinition, AsyncMigrationOperationSQL
+from posthog.client import sync_execute
+from posthog.constants import AnalyticsDBMS
+from posthog.models.cohort import Cohort
+
+"""
+Migration summary:
+Currently, behavioral cohorts are backed by cohortpeople and static cohorts are backed by person_static_cohort. The cohortpeople table has been
+significantly updated to support cohort versions and person versions/deletions. This migration will move person_static_cohort data and cohortpeople
+to a new table cohort_actors
+"""
+
+
+class Migration(AsyncMigrationDefinition):
+
+    description = "Move person_static_cohort and cohortpeople table to cohort_actors"
+
+    depends_on = "0007_persons_and_groups_on_events_backfill"
+
+    # After releasing this version we can remove code related to `person_static_cohort` table
+    posthog_max_version = "1.42.99"
+
+    # Only need to run if static cohort persons have been stored
+    def is_required(self):
+        static_rows_to_backfill_check = sync_execute(
+            """
+            SELECT 1
+            FROM person_static_cohort
+            LIMIT 1
+            """
+        )
+
+        rows_to_backfill_check = sync_execute(
+            """
+            SELECT 1
+            FROM cohortpeople
+            LIMIT 1
+            """
+        )
+
+        return len(static_rows_to_backfill_check) > 0 or len(rows_to_backfill_check) > 0
+
+    @cached_property
+    def operations(self):
```

Comment:
Probably makes sense to include the table creation in this migration? So we can have it one, vs. two different migrations that implicitly depend on each other